src/auto-evo/EvolutionaryTree.cs

```diff
@@ -1,21 +1,79 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 using AutoEvo;
 using Godot;
-using Godot.Collections;
-
+using Array = Godot.Collections.Array;
+
+/// <summary>
+///   An evolutionary tree showing species origins and mutations. Supports dragging and zooming.
+/// </summary>
+/// <remarks>
+///   <para>
+///     To use this, simply add this inside any controller and adjust its size.
+///     Don't put this under a draggable controller.
+///   </para>
+/// </remarks>
 public class EvolutionaryTree : Control
 {
-    private const float LEFT_MARGIN = 5.0f;
+    [Export]
+    public NodePath TimelinePath = null!;
+
+    [Export]
+    public NodePath TreePath = null!;
+
     private const float TIMELINE_HEIGHT = 50.0f;
     private const float TIMELINE_LINE_THICKNESS = 2.0f;
-    private const float TIMELINE_LINE_Y = 5.0f;
-    private const float TIMELINE_MARK_SIZE = 4.0f;
+    private const float TIMELINE_AXIS_Y = 5.0f;
+    private const float TIMELINE_MARK_LENGTH = 4.0f;
+    private const float TREE_LINE_THICKNESS = 4.0f;
+
+    /// <summary>
+    ///   Vertical separation between species.
+    /// </summary>
     private const float SPECIES_SEPARATION = 50.0f;
+
+    /// <summary>
+    ///   Horizontal separation between generations.
+    /// </summary>
     private const float GENERATION_SEPARATION = 100.0f;
+
+    /// <summary>
+    ///   The horizontal offset where species name is drawn to the right of the button or line
+    /// </summary>
     private const float SPECIES_NAME_OFFSET = 10.0f;
 
+    /// <summary>
+    ///   Each time the control is zoomed, sizeFactor gets multiplied (or divided) by this factor.
+    ///   This allows for smooth zooming, that zoom will not seem too quick when small, nor too slow when big.
+    /// </summary>
+    private const float ZOOM_FACTOR = 0.9f;
+
+    private const float SIZE_FACTOR_MIN = 0.2f;
+    private const float SIZE_FACTOR_MAX = 1.0f;
+
+    /// <summary>
+    ///   Draws outside this margin will be omitted to improve performance.
+    /// </summary>
+    private const float DRAW_MARGIN = 30.0f;
+
+    /// <summary>
+    ///   Default size of lato-small font.
+    /// </summary>
+    private const float SMALL_FONT_SIZE = 14.0f;
+
+    /// <summary>
+    ///   Default size of <see cref="EvolutionaryTreeNode"/>.
+    /// </summary>
+    private static readonly Vector2 TreeNodeSize = new(30, 30);
+
+    /// <summary>
+    ///   Auxiliary vector for <see cref="DRAW_MARGIN"/>
+    /// </summary>
+    private static readonly Vector2 DrawMargin = new(DRAW_MARGIN, DRAW_MARGIN);
+
+    // ReSharper disable 4 times RedundantNameQualifier
```

Comment:
I think it's better to use the disable and then enable comment rather than this "4 times" thing.